### DISCUSSION TITLE
Environment variables in .plzconfig

### DISCUSSION THREAD

**Discussion by PolarBear292208** [thought-machine/please#2221] [Q&A]
Is it possible to get access to environment variables in the `.plzconfig` file? To be more specific, I want to set the default build path to:

```
[build]
path = ${HOME}/.pyenv/shims:${HOMEBREW_PREFIX}/bin:/usr/local/bin:/usr/bin:/bin
```

and check it into our repository. I can replace ${HOME} with `~` and that works, but for Homebrew, it gets installed in different places on Intel vs M1 Macs so I need to reference a variable.

Alternatively, can you reference variables set in `.plzconfig` in a `.plzconfig.local` file, i.e. so you can append the default value?

**Comment by Tatskaari** [comment]
I'm afraid not. You can add the different config in `.plzconfig_darwin_amd64` and `.plzconfig_darwin_arm64` though. 

I might consider evaluating env variables in the path though. That could be handy.

**Reply by PolarBear292208** [reply]
That's a good idea, I tried it but I'm having some issues with how the different values are combined/overridden/ignored. What's the interaction of setting the same value in the different files? I'm finding there's some collision happening. My setup looks like:

.plzconfig
```
[build]
path = ~/.pyenv/shims:/usr/local/bin:/usr/bin:/bin
```

.plzconfig_darwin_arm64
```
[build]
path = ~/.pyenv/shims:/opt/homebrew/bin:/usr/local/bin:/usr/bin:/bin
```

If they're both set, plz seems to ignore them, e.g. it picks up the system installation of python. If I remove it from one file, it's happy again. Is there a way to see what value plz uses. It'd be great to have a command like `plz config --get build.path`.

**Reply by Tatskaari** [reply]
Oh.. it seems like it's layer the paths ontop of each other so you're protbably getting:

`~/.pyenv/shims:/usr/local/bin:/usr/bin:/bin:~/.pyenv/shims:/opt/homebrew/bin:/usr/local/bin:/usr/bin:/bin`

> It'd be great to have a command like plz config --get build.path.


There's `plz query config build.path` if you've updated to a newish version of Please ;P

**Reply by PolarBear292208** [reply]
> There's plz query config build.path if you've updated to a newish version of Please ;P

Cool, that's exactly what I wanted.

Even with the concatenated path, it should pick up the python in pyenv rather than the one in /usr/bin. I also tried just including the homebrew path in the darwin_arm64 file and got the same result.
